app/controllers/answers_controller.rb

```diff
@@ -0,0 +1,63 @@
+class AnswersController < ApplicationController
+  before_action :user_auth, except: %i[index]
+  before_action :set_answer, only: %i[update destroy]
+  before_action :validate_owner, only: %i[update destroy]
+  before_action :check_question_presence, only: %i[create update]
+
+  def index
+    render json: Answer.all, status: :ok
+  end
+
+  def create
+    answer = Answer.new(answer_params.merge(user_id: @current_user.id, question_id: answer_params[:question_id]))
+    return render json:  { message: answer.errors }, status: :unprocessable_entity unless answer.save
+
+    render json: answer, status: :created
+  end
+
+  def update
+    unless @answer.update(answer_params)
+      return render json: { message: @answer.errors.full_messages },
+                    status: :unprocessable_entity
+    end
+
+    render json: @answer, status: :ok
+  end
+
+  def destroy
+    unless @answer.destroy
+      return render json: { message: @answer.errors.full_messages },
+                    status: :unprocessable_entity
+    end
+
+    render json: @answer, status: :no_content
+  end
+
+  private
+
+  def check_question_presence
+    question = Question.find_by(id: answer_params[:question_id])
+    return render json: { message: 'question not found' }, status: :not_found if question.nil?
+  end
```

Comment:
no need to check for question presence. Question will always exist for the answer because you've set dependent destroy. So if the question gets deleted, the answer will be deleted as well.